genie-common/src/main/java/com/netflix/genie/common/dto/Command.java

```diff
@@ -93,9 +106,10 @@ protected Command(@Valid final Builder builder) {
     public static class Builder extends ExecutionEnvironmentDTO.Builder<Builder> {
 
         private final CommandStatus bStatus;
-        private final String bExecutable;
         private final long bCheckDelay;
+        private String bExecutable;
         private Integer bMemory;
+        private ImmutableList<String> bExecutableAndArguments;
```

Comment:
Not much point in making this an immutable list if you're allowing it to be overwritten. Patterns in other places in codebase is to make this `final List<String> bExecutableAndArguments = Lists.newArrayList()` and then `clear` and copy elements into it as needed then in the `build` copy into the concrete DTO `ImmutableList` instance.